documentation/advanced/tutorial-connection-indicator.asciidoc

```diff
@@ -0,0 +1,211 @@
+---
+title: The Connection Indicator
+order: 5
+layout: page
+---
+
+= The Connection Indicator
+
+The _connection indicator_ informs the user about the server connection status,
+and indicates the loading progress in case if network requests take some time.
+
+It shows connection status messages when the state of the server connection
+state changes, for example, if the user’s device goes offline and back online.
+Starting an application offline also triggers the connection status message.
+
+Also, to inform the user that loading is in progress and that the _UI_ is
+unresponsive, a _loading bar_ is displayed. A longer loading time might be due
+to, for example, bad network conditions. Vaadin automatically displays a loading
+indicator after a configurable delay when a server request starts, and hides
+it after the response processing has ended.
+
+*With default theme applied*, the connection indicator is at the top of
+the viewport. The connection status message shows after trigger for a short
+configurable duration. If the connection is available, the indicator containing
+then message then disappears, otherwise the indicator remains collapsed and
+expandable on hover.
+
+You can change messages text, delays, or customize the looks of the indicator.
+
+Vaadin automatically creates `vaadin-connection-indicator` and adds it to
+the body. The theming targets the loading bar with the `.v-loading-indicator`
+selector, and the connection status message with the `.v-status-message`, both
+are regular children inside the `vaadin-connection-indicator`. You can also
+toggle the default theming off for using custom styling of the indicator.
+
+[source,html]
+----
+<body>
+  <!-- application root level element omitted -->
+
+  <!--
+    The indicator element has none or one of the attributes:
+    - loading
+    - reconnecting
+    - offline
+  -->
+  <vaadin-connection-indicator>
+    <!-- "Vaadin removes "display: none" when indicator shown -->
+    <div class="v-loading-indicator first"
+     style="display: none;"></div>
+
+    <div class="v-status-message">
+      <span>
+        <!-- The current connection status message -->
+        Online
+      </span>
+    </div>
+  </vaadin-connection-indicator>
+</body>
+----
+
+== Configuring the component
+
+The connection indicator can be configured from TypeScript by accessing the
+component instance from `index.ts`:
+
+.index.ts (excerpt):
+[source,typescript]
+----
+// ...
+import {getConnectionIndicator} from '@vaadin/flow-frontend/ConnectionIndicator';
```

Comment:
A minor thing, we could use `import {getConnectionIndicator} from '@vaadin/flow-frontend/';` instead, since ConnectionIndicator is exported in the index.ts